index.js

```diff
@@ -10,8 +13,59 @@ const client = new Client({
     ]
 });
 
-client.on('ready', () => {
+const commandFiles = fs.readdirSync("./commands").filter(file => file.endsWith(".js"));
+const commands = [];
+client.commands = new Collection();
+
+for(const file of commandFiles) {
+	const command = require(`./commands/${file}`);
+	commands.push(command.data.toJSON());
+	client.commands.set(command.data.name, command);
+}
+
+client.once('ready', () => {
     console.info(`Logged in as ${client.user.tag}!`);
+
+	const rest = new REST({
+		version: "10"
+	}).setToken(process.env.BOT_TOKEN);
+
+	(async () => {
+		try {
+			if(process.env.ENV === "production") {
+				await rest.put(Routes.applicationCommands(process.env.CLIENT_ID), {
+					body: commands
+				});
+				console.log("Successfully registered commands globally.")
+			} else {
+				await rest.put(Routes.applicationGuildCommands(process.env.CLIENT_ID, process.env.GUILD_ID), {
+					body: commands
+				});
+				console.log("Successfully registered commands locally.")
+			}
+		} catch (err) {
+			if (err) console.error(err);
+		}
+	})();
+});
+
+client.on("interactionCreate", async interaction => {
+	if(!interaction.type === interaction.ApplicationCommand) return;
```

Comment:
This reads to me as "if does not have interaction type equals interaction application command return"

Can we do 	`if(interaction.type !== interaction.ApplicationCommand) return;
` instead